util/oec_utils.rb

```diff
@@ -28,4 +28,111 @@ def self.log_results(file, result, e = nil)
     end
   end
 
+  # Returns identifiers for a configured OEC test user
+  # @return [Hash]
+  def self.oec_user
+    {
+      first_name: @config['oec']['first_name'],
+      last_name: @config['oec']['last_name'],
+      uid: @config['oec']['uid'],
+      sis_id: @config['oec']['sis_id'],
+      email: @config['oec']['email']
+    }
+  end
+
+  # Returns the expected location of the folder containing sub-dirs and CSVs for the configured term code
+  # @return [String]
+  def self.term_folder
+    File.join(ENV['HOME'], "/OEC/#{@config['oec']['term']}")
+  end
+
+  # Returns departments that should be included in a given supervisors file but are missing
+  # @param file_name [String]
+  # @return [Array<String>]
+  def self.missing_supervisor_depts(file_name)
+    initial_csv = CSV.read(file_name, headers: true)
+    depts_present = []
+    initial_csv.each do |r|
+      1.upto(10) { |i| depts_present << r["DEPT_NAME_#{i}"] if r["DEPT_NAME_#{i}"]}
+    end
+    depts_present.compact!.uniq!
+    depts_needed = OECDepartments::DEPARTMENTS.map { |d| d.form_code }
+    depts_needed.compact!.uniq!
```

Comment:
@pfarestveit A gotcha with chaining exclamation-point methods is that they can return nil if no changes are made. 

https://ruby-doc.org/core-1.9.3/Array.html#method-i-compact-21

So in the case where `depts_needed` has no values to clear out, `depts_needed.compact!` will return nil, and calling `uniq!` will return an error.

`depts_present.compact!.uniq!` should be safe to change to `depts_present.uniq!`, since you're now doing a nil check while building it.

For `depts_needed`, I'd break `depts_needed.compact!` and `depts_needed.uniq!` into separate lines.